mantidproject/mantid
# Sum banks for 2D detectors

**Description of work.**
This PR ensures the `ReflectomatryISISLoadAndProcess` algorithm sums the banks of 2D detectors before performing a reduction.

**To test:**

Run this and check the output says that it matches the summed workspace (not the un-summed one).
```
ReflectometryISISPreprocess(InputRunList='INTER45455_inst', OutputWorkspace='TOF_45455')

# With sum-banks step
ReflectometryISISSumBanks(InputWorkspace='TOF_45455', OutputWorkspace='TOF_45455_summed')
ReflectometryReductionOneAuto(InputWorkspace='TOF_45455_summed', ProcessingInstructions='22-27', ThetaIn=0.5, MomentumTransferMin=0.0064527921283059177, MomentumTransferStep=0.02017069966273068, MomentumTransferMax=0.072643025200858427, ScaleFactor=1, OutputWorkspaceBinned='IvsQ_binned_45455_summed', OutputWorkspace='IvsQ_45455_summed')

# Without sum-banks step
ReflectometryReductionOneAuto(InputWorkspace='TOF_45455', ProcessingInstructions='22-27', ThetaIn=0.5, MomentumTransferMin=0.0064527921283059177, MomentumTransferStep=0.02017069966273068, MomentumTransferMax=0.072643025200858427, ScaleFactor=1, OutputWorkspaceBinned='IvsQ_binned_45455_unsummed', OutputWorkspace='IvsQ_45455_unsummed', OutputWorkspaceWavelength='__TMP0x55838730ba00')

# Via workflow algorithm
ReflectometryISISLoadAndProcess(InputRunList='INTER45455_inst', ProcessingInstructions='22-27', ThetaIn=0.5, OutputWorkspaceBinned='IvsQ_binned_45455_workflow', OutputWorkspace='IvsQ_45455_workflow', OutputWorkspaceWavelength='__TMP0x55838730ba00')
match_summed, _ = CompareWorkspaces('IvsQ_binned_45455_workflow', 'IvsQ_binned_45455_summed')
match_unsummed, _ = CompareWorkspaces('IvsQ_binned_45455_workflow', 'IvsQ_binned_45455_unsummed')
print('Matches summed workspace = ', match_summed)
print('Matches unsummed workspace = ', match_unsummed)
```

```
Matches summed workspace =  True
Matches unsummed workspace =  False
```

Also try running with the property `ROIDetectorIDs='2060-2100'` in ReflectometryISISSumBanks and ReflectometryISISLoadAndProcess above and check the outputs still match.

Fixes #34109 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
